api/src/ginger_calls.rs

```diff
@@ -684,6 +684,95 @@ pub fn get_lazy_ginger_smt_path(tree: &mut LazyGingerSMT, leaf_position: u64) ->
     tree.get_merkle_path(Coord::new(0, leaf_position as usize))
 }
 
+////////////BITVECTOR BASED ON SPARSE MERKLE TREE
+
+pub type BitvectorTree = GingerSMT;
+
+pub fn get_bvt(height: usize, db_path: &str) -> Result<BitvectorTree, Error>{
+    get_ginger_smt(height, db_path)
+}
+
+pub fn flush_bvt(tree: &mut BitvectorTree) {
+    flush_ginger_smt(tree)
+}
+
+pub fn set_bvt_persistency(tree: &mut BitvectorTree, persistency: bool) {
+    set_ginger_smt_persistency(tree, persistency)
+}
+
+pub fn get_bvt_root(tree: &BitvectorTree) -> FieldElement {
+    get_ginger_smt_root(tree)
+}
+
+pub fn get_bvt_path(tree: &mut BitvectorTree, bit_position: u64) -> GingerMHTPath {
+    get_ginger_smt_path(tree, bit_position / bits_per_bvt_leaf())
+}
+
+pub fn get_bvt_leaf(tree: &BitvectorTree, bit_position: u64) -> Option<FieldElement> {
+    get_leaf_in_bvt(tree, bit_position / bits_per_bvt_leaf())
+}
+
+pub fn set_bvt_bit(tree: &mut BitvectorTree, bit_position: u64){
+    modify_bit_in_bvt(tree, bit_position, true)
+}
+
+pub fn reset_bvt_bit(tree: &mut BitvectorTree, bit_position: u64){
+    modify_bit_in_bvt(tree, bit_position, false)
+}
+
+pub fn get_bvt_bit(tree: &BitvectorTree, bit_position: u64) -> bool {
+    let leaf_bitsize = bits_per_bvt_leaf();
+
+    let leaf_index = bit_position / leaf_bitsize;
+    if let Some(leaf) = get_leaf_in_bvt(tree, leaf_index) {
+        let leaf_bits= leaf.write_bits();
+        assert_eq!(leaf_bitsize, (leaf_bits.len() - 1) as u64);
+        leaf_bits[(leaf_bitsize - (bit_position % leaf_bitsize)) as usize]
+    } else { // leaf doesn't even exist, so the bit isn't set
+        false
+    }
+}
+
+fn bits_per_bvt_leaf() -> u64 {
+    (FieldElement::size_in_bits() - 1) as u64 // using one bit lesser to make leafs never exceed the field modulus
+}
+
+fn get_leaf_in_bvt(tree: &BitvectorTree, leaf_position: u64) -> Option<FieldElement> {
+    tree.get_leaf(Coord::new(0, leaf_position as usize))
```

Comment:
Yes, this would be an easier approach, but as I understood from implementation of the insert_leaf method in BMT, there is a check for presence of a specified node before retrieving it from database:
 if self.state.present_node.contains(&coord) {
            let old_hash = self.get_from_db(coord.idx).unwrap();

https://github.com/HorizenOfficial/ginger-lib/blob/0dc2e545aa5b1b9ce291c713ad47f99f654bc8ca/primitives/src/merkle_tree/field_based_mht/smt/big_merkle_tree.rs#L411

So to be consistent with BMT implementation I proposed additional method with this check and further fetching from db. But if this check for presence is not necessarry and direct access to DB in case if node is not actually present will not cause unexpected errors then we can just directly use the get_from_db method.

Reply:
Yes, because the present_node is stored in memory, so I guess it's a quick way to avoid looking on disk if you already know that the leaf is not present. Okey, let's proceed with the get_leaf() function !